tests/gh-actions/test_dex_auth.sh

```diff
@@ -0,0 +1,275 @@
+
+# Verify Dex secret is properly set
+echo "Checking if Dex password is set..."
+if ! kubectl get secret -n auth dex-secret > /dev/null 2>&1; then
+  echo "Creating Dex password secret..."
+  # The default password in the test script is 12341234
+  kubectl create secret generic dex-secret -n auth --from-literal=DEX_USER_PASSWORD=$(python3 -c 'from passlib.hash import bcrypt; print(bcrypt.using(rounds=12, ident="2y").hash("12341234"))')
+  # Restart Dex if it exists
+  if kubectl get deployment -n auth dex &>/dev/null; then
+    kubectl rollout restart deployment -n auth dex
+    kubectl wait --for=condition=Available deployment -n auth dex --timeout=180s || echo "Dex deployment not available, but continuing"
+  fi
+fi
+
+# Create or update the ConfigMap for Dex configuration with correct redirectURIs
+echo "Ensuring Dex ConfigMap has correct redirectURIs..."
+kubectl get cm -n auth dex -o yaml > /tmp/dex-cm.yaml
+if grep -q "redirectURIs.*http://authservice" /tmp/dex-cm.yaml; then
+  echo "Fixing redirect URI in Dex ConfigMap..."
+  # Create updated ConfigMap with correct redirectURIs
+  cat > /tmp/dex-config.yaml << EOF
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: dex
+  namespace: auth
+data:
+  config.yaml: |
+    issuer: http://dex.auth.svc.cluster.local:5556/dex
+    storage:
+      type: kubernetes
+      config:
+        inCluster: true
+    web:
+      http: 0.0.0.0:5556
+    logger:
+      level: "debug"
+      format: text
+    oauth2:
+      skipApprovalScreen: true
+    enablePasswordDB: true
+    staticPasswords:
+    - email: user@example.com
+      hashFromEnv: DEX_USER_PASSWORD
+      username: user
+      userID: "15841185641784"
+    staticClients:
+    # https://github.com/dexidp/dex/pull/1664
+    - idEnv: OIDC_CLIENT_ID
+      redirectURIs: ["/oauth2/callback"]
+      name: 'Dex Login Application'
+      secretEnv: OIDC_CLIENT_SECRET
+EOF
+  kubectl apply -f /tmp/dex-config.yaml
+  
+  # Restart Dex to pick up the new config
+  kubectl rollout restart deployment -n auth dex
+  sleep 10
+fi
+
+# Create OIDC client secrets needed for authentication
+echo "Creating Dex client secrets..."
+kubectl create secret generic oidc-client-secret -n auth \
+  --from-literal=OIDC_CLIENT_ID=kubeflow-oidc-authservice \
+  --from-literal=OIDC_CLIENT_SECRET=pUBnBOY80SnXgjibTYM9ZWNzY2xreNGQok \
+  --dry-run=client -o yaml | kubectl apply -f -
+
+kubectl create secret generic oidc-client-secret -n oauth2-proxy \
+  --from-literal=OIDC_CLIENT_ID=kubeflow-oidc-authservice \
+  --from-literal=OIDC_CLIENT_SECRET=pUBnBOY80SnXgjibTYM9ZWNzY2xreNGQok \
+  --dry-run=client -o yaml | kubectl apply -f -
+
+# Update Dex deployment with env variables if needed
+echo "Ensuring Dex deployment has correct environment variables..."
+if ! kubectl get deploy -n auth dex -o yaml | grep -q "OIDC_CLIENT_ID"; then
+  cat > /tmp/dex-patch.yaml << EOF
+spec:
```

Comment:
Why are you copying the manifests? see https://github.com/kubeflow/manifests/blob/master/.github/workflows/dex_oauth2-proxy_test.yaml